hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/MapNearCacheManager.java

```diff
@@ -134,13 +134,14 @@ public void shutdown() {
      */
     @Override
     public boolean destroyNearCache(String mapName) {
-        if (!super.destroyNearCache(mapName)) {
-            return false;
-        }
-
         String uuid = nodeEngine.getLocalMember().getUuid();
-        invalidator.destroy(mapName, uuid);
-        return true;
+        invalidator.invalidateAllKeys(mapName, uuid);
```

Comment:
The `Invalidator` clears the Near Cache on all listening nodes, e.g. clients. You don't have to have a local Near Cache for the need of this call. Otherwise, if there are no listener registrations, this call will do nothing:
```java
    public final void invalidateAllKeys(String dataStructureName, String sourceUuid) {
        assert dataStructureName != null;
        assert sourceUuid != null;

        int orderKey = getPartitionId(dataStructureName);
        Invalidation invalidation = newClearInvalidation(dataStructureName, sourceUuid);
        sendImmediately(invalidation, orderKey);
    }

    protected final void sendImmediately(Invalidation invalidation, int orderKey) {
        String dataStructureName = invalidation.getName();

        Collection<EventRegistration> registrations = eventService.getRegistrations(serviceName, dataStructureName);
        for (EventRegistration registration : registrations) { // <------ will do nothing on empty registration 
            if (eventFilter.apply(registration)) {
                eventService.publishEvent(serviceName, registration, invalidation, orderKey);
            }
        }
    }
```